src/yaml/from_yaml.cr

```diff
@@ -48,6 +48,18 @@ private def parse_scalar(ctx, node, type : T.class) forall T
   end
 end
 
+module Iterator(T)
+  # Reads the content for an iterator from a YAML sequence
+  def self.new(ctx : YAML::ParseContext, node : YAML::Nodes::Node)
```

Comment:
I'm wondering if adding this method for YAML actually makes sense, if it can't be lazy. It's pretty much the same as `Array.from_yaml`.

Reply:
I added it just for completeness. I was also wondering if it's required. If you prefer a PR without this, I will remove it.

Reply:
I think I'd prefer to leave it. It's easier to add later if necessary, than to remove it.
Let's wait for other opinions, though.

Reply:
I removed it and also squashed all commits.

Reply:
Please don't do force pushes (see https://github.com/crystal-lang/crystal/blob/master/CONTRIBUTING.md#minimum-requirements).

Reply:
Sorry for that.